### PR TITLE
fix: ignore optional non resolved dependencies

### LINKED ISSUE #54: Optional unresolved dependencies should not be included
### What happened?

Optional unresolved dependencies should not be included. This is causing problems when asking Snyk and also the report will show dependencies that are not really used.

When a dependency has optional dependencies, if such dependencies are included directly from the project of from other dependencies, then they become effective.

### Please provide runtime information.

main branch

### Relevant log output

```shell
Example:

"jsdom": {
      "version": "19.0.0",
      "resolved": "https://registry.npmjs.org/jsdom/-/jsdom-19.0.0.tgz",
      "overridden": false,
      "dependencies": {
        "acorn": {
          "version": "8.10.0",
          "resolved": "https://registry.npmjs.org/acorn/-/acorn-8.10.0.tgz",
          "overridden": false
        },
        "canvas": {},
        ...
}
```
```


### How can this issue be reproduced?

The library jsdom@19.0.0 has an optional dependency with canvas

### PR DESCRIPTION
**Pull request by ruromero**
## Description

Ignore optional dependencies.

**Related issue (if any):** fixes #54 

## Checklist

- [x] I have followed this repository's contributing guidelines.
- [x] I will adhere to the project's code of conduct.

### CODE REVIEW

File: test/providers/tst_manifests/npm/package_json_deps_with_exhortignore_object/stack_expected_sbom.json [hunk 1]
```diff
@@ -1,1870 +1,2533 @@
...
-			"ref": "pkg:npm/undefsafe@2.0.5",
-			"dependsOn": []
-		}
-	]
-}
+    "bomFormat": "CycloneDX",
+    "specVersion": "1.4",
+    "version": 1,
+    "metadata": {
+        "timestamp": "2023-09-20T14:27:17.520Z",
```
Comment (zvigrinberg) [inline test/providers/tst_manifests/npm/package_json_deps_with_exhortignore_object/stack_expected_sbom.json:6]: @ruromero 
Just rename the timestamp back so tests will succeed.
Thanks.

**Review by zvigrinberg — APPROVED** [review]
@ruromero Thanks for the catch!
Just Please changed the ts accordingly so tests will not fail.

File: test/providers/tst_manifests/npm/package_json_deps_with_exhortignore_object/stack_expected_sbom.json (hunk 1 above)
Comment (ruromero) [reply test/providers/tst_manifests/npm/package_json_deps_with_exhortignore_object/stack_expected_sbom.json]: Oh, I missed it

**Review by zvigrinberg — APPROVED** [review]
lgtm

**Pull request merged**